test/Main.hs

```diff
@@ -0,0 +1,46 @@
+module Main where
+
+import           Test.Tasty (after, defaultMain, testGroup, DependencyType(AllFinish), TestTree)
+
+import           BasePrelude
+
+import qualified Fencer.Rules.Test as R
+import qualified Fencer.Server.Test as S
+import qualified Fencer.Types.Test as T
+
+
+tests :: TestTree
+tests = testGroup "All tests"
+  [ types
+  , rules
+  -- 'after' is needed to avoid running the 'rules' and 'server' tests
```

Comment:
Is this the reason why you got rid of `tasty-discover`? If yes, perhaps let's instead disable the parallelism by running tests with `+RTS -N1 -RTS`.

Reply:
Right, I couldn't set the order of test execution when `tasty-discover` was used.

If I am not missing anything, `+RTS -N1 -RTS` would mean the program hangs because it would not be able to fork a thread and start a gRPC server.

Reply:
@neongreen 

> If yes, perhaps let's instead disable the parallelism by running tests with `+RTS -N1 -RTS`.

I don't think it's a good idea as it solves a pretty much already solved problem, but also makes tests slower.

@mdimjasevic 

> If I am not missing anything, `+RTS -N1 -RTS` would mean the program hangs because it would not be able to fork a thread and start a gRPC server.

Threads are forkable with `-threaded` even with a single core.